src/LibDeclarations/webxr.nativeextensions.d.ts

```diff
@@ -1,11 +1,82 @@
-// This file contains native only extensions for WebXR  These APIs are not supported in the browser yet.
+// This file contains native only extensions for WebXR. These APIs are not supported in the browser yet.
 // They are intended for use with either Babylon Native https://github.com/BabylonJS/BabylonNative or
 // Babylon React Native: https://github.com/BabylonJS/BabylonReactNative
 
+// Open question: Do better types exist to use for vectors/quaternions?
+interface XRVector {
+    x: number;
+    y: number;
+    z: number;
+}
+
+interface XRQuaternion {
+    x: number;
+    y: number;
+    z: number;
+    w: number;
+}
+
+// Open question: Should XRGeometryType be defined or should all types be strings?
+type XRGeometryType = "unknown" | "background" | "wall" | "floor" | "ceiling" | "platform";
+
+interface XRFieldOfView {
+    angleLeft: number;
+    angleRight: number;
+    angleUp: number;
+    angleDown: number;
+}
+
+interface XRFrustum {
+    position: XRVector;
+    rotation: XRQuaternion;
+    fieldOfView: XRFieldOfView;
+    farDistance: number;
+}
+
+interface XRPlane {
+    // Open question: Should geometry id's and types be declared on XRPlanes or queried through other means?
+    geometryId?: number;
+    geometryType?: XRGeometryType | string;
+}
+
+// Open question: Should XRMesh have a concept of a XRReferenceSpace similar to XRPlane?
+interface XRMesh {
+    positions: Float32Array;
+    indices: Uint32Array;
+    normals?: Float32Array;
+    // Open question: Do we need lastChangedTime declared for XRMeshes?
+    lastChangedTime: number;
+    // Open question: Should geometry id's and types be declared on XRMeshes or queried through other means?
+    geometryId?: number;
+    geometryType?: XRGeometryType | string;
+}
+
+type XRDetectionBoundaryType = "frustum" | "sphere" | "box";
+
+interface XRDetectionBoundary {
+    isStationary?: boolean;
+    type?: XRDetectionBoundaryType;
+    frustum?: XRFrustum;
+    sphereRadius?: number;
+    boxDimensions?: XRVector;
+}

```

Comment:
Since the intention here is that it is one of the three types of boundaries, we should type this more strongly:
```suggestion
interface XRFrustrumDetectionBoundary {
  type: "frustum";
  frustum: XRFrustum;  
}

interface XRSphereDetectionBoundary {
  type: "sphere";
  radius: number;
}

interface XRBoxDetectionBoundary {
  type: "box";
  extent: XRVector;
}

type XRDetectionBoundary = XRFrustrumDetectionBoundary | XRSphereDetectionBoundary | XRBoxDetectionBoundary;
```
This will make intellisense help you do the right thing when constructing a boundary or reading a boundary. For example:
```ts
function Test(boundary: XRDetectionBoundary) {
  if (boundary.type === "frustum") {
    let f = boundary.frustum;
  }
}
```
Here, inside the `if` block where you have already determined that `type` is `"frustum"`, then `boundary` has been type narrowed and intellisense and the compiler know that the object has a property named `frustum` of type `XRFrustum`.

Reply:
@ryantrem  I won't claim to be a typescript expert, but is it better to have interface inheritance here where type is guaranteed in XRDetectionBoundary. Then other detection boundaries extend XRDetectionBoundary. Or is it better to have a type definition as described above? It seems like with the type definition we could accidentally end up with types added to the type definition that don't populate a type field.

Reply:
I think the union type I suggest above is a stronger type assertion than having a base interface with additional interfaces that extend it, so I would recommend that route.